src/formats/apple-notes/convert-note.ts

```diff
@@ -0,0 +1,398 @@
+];
+
+export class NoteConverter extends ANConverter {
+	note: ANNote;
+	
+	listNumber = 0;
+	listIndent = 0;
+	multiRun = ANMultiRun.None;
+	
+	static protobufType = 'ciofecaforensics.Document';
+	
+	constructor(importer: AppleNotesImporter, document: ANDocument | ANTableObject) {
+		super(importer);
+		this.note = document.note;
+	}
+
+	parseTokens(): ANFragmentPair[] {	
+		let i = 0;
+		let offsetStart = 0;
+		let offsetEnd = 0;
+		let tokens = [];
+		
+		while (i < this.note.attributeRun.length) {
+			let attr: ANAttributeRun;
+			let attrText = '';
+			let nextIsSame = true;
+			
+			/* First, merge tokens with the same attributes */
+			do {
+				attr = this.note.attributeRun[i];
+				offsetEnd = offsetEnd + attr.length;
+				attrText += this.note.noteText.substring(offsetStart, offsetEnd);
+				
+				offsetStart = offsetEnd;
+				nextIsSame = (i == this.note.attributeRun.length - 1)
+					? false
+					: attrEquals(attr, this.note.attributeRun[i+1]);
+				
+				i++;
+			}
+			while (nextIsSame);
+			
+			/* Then, since Obsidian doesn't like formatting crossing new lines or 
+			starting/ending at spaces, divide tokens based on that */
+			for (let fragment of attrText.split(FRAGMENT_SPLIT)) {
+				if (!fragment) continue;
+				tokens.push({ attr, fragment }); 
+			}
+		}
+		
+		return tokens;
+	}
+		
+	async format(table = false): Promise<string> {
+		let fragments = this.parseTokens();
+		let firstLineSkip = !table && this.importer.omitFirstLine;
+		let converted = '';
+		
+		for (let j = 0; j < fragments.length; j++) {		
+			let { attr, fragment } = fragments[j];
+			
+			if (firstLineSkip) {
+				if (fragment.contains('\n') || attr.attachmentInfo) firstLineSkip = false;
+				else continue;
+			}
+
+			attr.fragment = fragment;		
+			attr.atLineStart = j == 0 ? true : fragments[j - 1]?.fragment.contains('\n');
+			
+			converted += this.formatMultiRun(attr);
+			
+			if (!/\S/.test(attr.fragment) || this.multiRun == ANMultiRun.Monospaced) {
+				converted += attr.fragment;
+			}
+			else if (attr.attachmentInfo) {
+				converted += await this.formatAttachment(attr);
+			}
+			else if (attr.superscript || attr.underlined || attr.color || attr.font || this.multiRun == ANMultiRun.Alignment) {
+				converted += this.formatHtmlAttr(attr);
+			}
+			else {
+				converted += this.formatAttr(attr);
+			}
+		}
+		
+		if (this.multiRun != ANMultiRun.None) converted += this.formatMultiRun(null);
+		if (table) converted.replace('\n', '<br>').replace('|', '&#124;');
+		
+		return converted.trim();
+	}
+	
+	/** Format things that cover multiple ANAttributeRuns. */
+	formatMultiRun(attr: ANAttributeRun | null): string {
+		const styleType = attr?.paragraphStyle?.styleType;
+		let prefix = '';
+		
+		switch (this.multiRun) {
+			case ANMultiRun.List:
+				if (
+					(attr?.paragraphStyle?.indentAmount == 0 && 
+					!LIST_STYLES.includes(styleType!)) ||
+					isBlockAttachment(attr!)
```

Comment:
Ah it seems like this is asserted to be not null here - it probably shouldn't?

Reply:
thanks for catching this - iirc it was initially never going to be null and then I changed that later on but forgot to update here